### PR TITLE
Convert DE to family

### PR DESCRIPTION
**Pull request by jrapin**
## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context / Related issue

#100 
This instantiates all DE algorithms through their family class. It also make OptimizerFamily instances compatible with benchmarks.

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/optimization/differentialevolution.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def __init__(self, dimension: int, budget: Optional[int] = None, num_workers: in
         self._lhs_init = False
         self._qr_init = False
         self._por_DE = False
+        self._recommendation = "optimistic"
```
Comment (teytaud) [inline nevergrad/optimization/differentialevolution.py]: maybe this should be pessimistic.
In most cases this does not change anything (DE is continuous, all points are distinct) but it's cleaner (intuitively recommendations are pessimistic).

File: nevergrad/optimization/differentialevolution.py [hunk 2]
```diff
@@ -163,76 +167,43 @@ def _internal_tell(self, x: ArrayLike, value: float) -> None:
...
+
+OnePointDE = DifferentialEvolution(crossover=1).with_name("OnePointDE", register=True)
+TwoPointsDE = DifferentialEvolution(crossover=2).with_name("TwoPointsDE", register=True)
+LhsDE = DifferentialEvolution(lhs_init=True).with_name("LhsDE", register=True)
+QrDE = DifferentialEvolution(qr_init=True).with_name("QrDE", register=True)
+MiniDE = DifferentialEvolution(scale="mini").with_name("MiniDE", register=True)
+MiniLhsDE = DifferentialEvolution(lhs_init=True, scale="mini").with_name("MiniLhsDE", register=True)
+MiniQrDE = DifferentialEvolution(qr_init=True, scale="mini").with_name("MiniQrDE", register=True)
+OnePointDE = DifferentialEvolution(recommendation="noisy").with_name("NoisyDE", register=True)
+AlmostRotationInvariantDE = DifferentialEvolution(CR=.9).with_name("AlmostRotationInvariantDE", register=True)
```
Comment (teytaud) [inline nevergrad/optimization/differentialevolution.py]: maybe you can plot the illcond* before and after for checking that it's not worst ?

looks clean to me :-)

Comment (jrapin) [reply nevergrad/optimization/differentialevolution.py]: no visible change + recommendation record is unchanged for tests

File: nevergrad/optimization/differentialevolution.py (hunk 1 above)
Comment (jrapin) [reply nevergrad/optimization/differentialevolution.py]: Changed to pessimistic

Comment (jrapin) [reply nevergrad/optimization/differentialevolution.py]: Reverted to optimistic because pessimistic deteriorates results

**Pull request merged**